README.md

```diff
@@ -1,63 +1,103 @@
-    - Creates RLS policies.
-    - Creates users.
-    - Detects and skips duplicates (on Superset) for roles, users and RLS policies.
-    - User details missing usernames and passwords are auto-generated.
-    - Creates CSV file with user details.
-
-This script can be run multiple times without worry of duplicates.
-
-### Important Notes
-- Always test roles and RLS policies in a non-production environment first
-- Regularly audit user permissions and access levels
-- Back up existing configurations before making bulk changes
-- Monitor API rate limits when creating multiple entities
+# Superset User Management
 
+This project provides a set of functions to help you manage users, roles and permissions in Superset.
 
+## Installation
+
+Clone the repository and install the dependencies:
+```bash
+npm install
+```
+
+
+# Usage
+
+It is recommended to run the application locally and test using a local Superset instance before working on the production instance.
+A REST API is available at `{{superset_url}}/swagger/v1`. To access the security endpoints that allow you to manage roles, add the 
+following to your Superset config file:
+
+`FAB_ADD_SECURITY_API = True`
+
+Restart the Superset container for the setting to take effect. You should now have access to the security endpoints.
+
+### Authentication
+
+An admin account is required to make any requests to the Superset API. The username and password can be configured in the .env file, along with the Superset URL. 
+
+`POST` and `PUT` requests require an additional header: `X-CSRFToken` which can be obtained by logging in to Superset, then visiting the `/security/csrf_token/` endpoint with the access token in the `Authorization` header.
+
+### Roles
+
+These define what datasets (tables) and permissions a user has access to. Roles are meant to be re-usable and can be re-assigned when users are moved between places. 
+
+### Row-level security
+
+These define what datasets and roles restrictions apply to. For example, a CHA role should have access to data from their CHU only. 
+As such, The RLS policy contains a clause e.g chu_code = `{{chu_code}}`. The query builder will replace this with the CHU code of the user and append it to the select query. The same applies to sub-county and county users. RLS policies only apply to roles.
+
+### Users
+
+Users are assigned roles which in turn have the requisite RLS policies attached. User accounts are to be deactivated by setting toggling the `active` status on the UI or setting `"active": false` in the payload.
+
+
+## Creating Roles
+
+We will be dealing with 3 main types of roles:
+- County
+- Sub-county
+- CHA
+
+Create these roles first, either through the UI or by making a `POST` request to the endpoint `/security/roles/` with the following JSON payload:
+
+```json
+{
+    "name": "role-name"
+}
+```
+
+Next, update the permissions for the roles. Subsequent roles for individual types will be copied off of these `base` roles. 
```

Comment:
Question: should we describe how to update the permissions for the roles on the UI ? Or via API ?